debezium-reactive/src/main/java/io/debezium/reactive/Converter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.reactive;
+
+import java.util.Map;
+
+import org.apache.kafka.connect.source.SourceRecord;
+
+public interface Converter<T> {
```

Comment:
this thing is almost like the one from Kafka connect (expect for slightly different signatue and a `toConnectData`) - even some of the javadoc is "copied" :-) 

I understand that it's mainly for _converting_ the output streaming events, to types... but I wonder if that thing could be named better. 

Another thing is that, for an API, this might be (at some point) independent from kafka (meaning this directly works on `SourceRecord`)

Reply:
The current idea is to make the abstraction from `SourceRecord` et al. only at the level of clients of the reactive APIs. So they'd set up the stream with some type identifier (pseudo code):

    Stream<String> = getReactiveEngine(AsJson.class).build();

`AsJson` would make the link to the right converter and be parameterized with the target type (e.g. String in case of JSON). That way the consumer would be shielded away from `SourceRecord` (unless when explicitly using the `AsSourceRecord` no-ops converter), while we can keep the inner workings of Debezium for now.

Would that make sense?

Reply:
Hmm, I like the naming convention proposed by Gunnar. So what if I rename interface `Converter` to `AsType` and `JsonConverter` class to `AsJson`?

Reply:
Note I had one more indirection in mind. I.e. `AsType` wouldn't represent the converter itself nor would it have a link to the same (to prevent unconditional exposure of `SourceRecord` in the API). What I thought of was this:

    public interface AsType<T> {
        Class<T> getTargetType();
    }

    public class AsJson implements AsType<String> {
        Class<String> getTargetType() {return String.class; };
    }

So this would just be used as to get a handle to the type into the API invocation chain and allow to return a properly typed `ReactiveEngine<String>`.

Then, in the implementation code we could for instance look up suitable converters using the `ServiceLoader` mechanism. For that, `Converter` would have a method `<T extends AsType> getConvertedType()`. We'd look up all converters registered  via _META-INF/services_ and chose the one matching incoming `AsType`. If there are multiple converters for the same `AsType` sub-class, a boostrap error would be raised.

This gives us both: type safe access for users in the API, while not exposing the KC API. Unless a user would choose `AsSourceRecord.class` of course, but that's a deliberate choice on their end then.